From b7c120787cc9858826aed9ad6d39e287d333d395 Mon Sep 17 00:00:00 2001
From: Xulin Sun <xulin.sun@windriver.com>
Date: Wed, 26 Feb 2014 16:35:58 +0800
Subject: [PATCH 3/3] MMC/eSDHC: add .get_cd callback for sdhci_do_get_cd

To check the mmc card's status, some vendors have their private
implementation, here add the callback in the sdhci_do_get_cd.

Signed-off-by: Xulin Sun <xulin.sun@windriver.com>
---
 drivers/mmc/host/sdhci.c |    9 +++++++++
 1 files changed, 9 insertions(+), 0 deletions(-)

diff --git a/drivers/mmc/host/sdhci.c b/drivers/mmc/host/sdhci.c
index 2f72119..d215cb8 100644
--- a/drivers/mmc/host/sdhci.c
+++ b/drivers/mmc/host/sdhci.c
@@ -1646,6 +1646,15 @@ static void sdhci_set_ios(struct mmc_host *mmc, struct mmc_ios *ios)
 static int sdhci_do_get_cd(struct sdhci_host *host)
 {
 	int gpio_cd = mmc_gpio_get_cd(host->mmc);
+	unsigned long flags;
+	int present;
+
+	if (host->ops->get_cd) {
+		spin_lock_irqsave(&host->lock, flags);
+		present = host->ops->get_cd(host);
+		spin_unlock_irqrestore(&host->lock, flags);
+		return present;
+	}
 
 	if (host->flags & SDHCI_DEVICE_DEAD)
 		return 0;
-- 
1.7.5.4

